worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala

```diff
@@ -510,22 +510,30 @@ private[celeborn] class Worker(
           // During graceful shutdown, to avoid allocate slots in this worker,
           // add this worker to master's blacklist. When restart, register worker will
           // make master remove this worker from blacklist.
-          rssHARetryClient.send(ReportWorkerUnavailable(List(workerInfo).asJava))
-          val interval = conf.checkSlotsFinishedInterval
-          val timeout = conf.checkSlotsFinishedTimeoutMs
-          var waitTimes = 0
-
-          def waitTime: Long = waitTimes * interval
+          try {
+            rssHARetryClient.askSync(
+              ReportWorkerUnavailable(List(workerInfo).asJava),
+              OneWayMessageResponse.getClass)
+
+            val timeout = conf.checkSlotsFinishedTimeoutMs
+            val startTime = System.currentTimeMillis()
+            def waitTime: Long = System.currentTimeMillis() - startTime
+            val interval = conf.checkSlotsFinishedInterval
+            while (!partitionLocationInfo.isEmpty && waitTime < timeout) {
+              Thread.sleep(interval)
+            }
 
-          while (!partitionLocationInfo.isEmpty && waitTime < timeout) {
-            Thread.sleep(interval)
-            waitTimes += 1
-          }
-          if (partitionLocationInfo.isEmpty) {
-            logInfo(s"Waiting for all PartitionLocation released cost ${waitTime}ms.")
-          } else {
-            logWarning(s"Waiting for all PartitionLocation release cost ${waitTime}ms, " +
-              s"unreleased PartitionLocation: \n$partitionLocationInfo")
+            if (partitionLocationInfo.isEmpty) {
+              logInfo(s"Waiting for all PartitionLocation released cost ${waitTime}ms.")
+            } else {
+              logWarning(s"Waiting for all PartitionLocation release cost ${waitTime}ms, " +
+                s"unreleased PartitionLocation: \n$partitionLocationInfo")
+            }
+          } catch {
+            case e: Throwable =>
+              logError(
+                s"Fail report to master, unreleased PartitionLocation: \n$partitionLocationInfo",
+                e)
```

Comment:
The try catch should only wrapper the `sendSync`, sendSync fail should not stop the graceful process.